### ISSUE TITLE
NglParseError for Linting Declarative Pipeline Inconsistency

### ISSUE THREAD

**Issue by mwallace72** [nvuillam/npm-groovy-lint#120]
Hi there, thanks for the tool.

I'm using: 
npm-groovy-lint@7.6.4
npm@6.13.4
node@v12.14.1

I have an assortment of declarative pipeline files.
I have made a `.groovylintrc.json` file at the root of my project with the following contents: 
```json
{
    "extends": "recommended-jenkinsfile",
    "rules": {
      "formatting.Indentation": {
          "spacesPerIndentLevel": 2,
          "severity": "warning"
      }
    }
}
```

When running  `npm-groovy-lint  --files "**/*Jenkinsfile" --fix` I get a mix of output.

Some of my files are flagged for normal things (expected, as I just began using this tool) like line length, etc. However, some of my files are tagged with things like

`error    Unexpected input: '{' @ line 1, column 10.  NglParseError` (they all reference column 10)

The file in question begins with 
```
pipeline {
....
}
``` 
as a declarative pipeline ought to.

Oddly enough, not _all_ of my declarative pipelines are so flagged - some do not error at all. I am unclear as to the cause of this.

**Comment by nvuillam** [comment]
Please could you post your jenkinsfiles so I can reproduce ? :) 
If they can not be public, you can mail them

**Comment by mwallace72** [comment]
Reproduced with this content:
```
pipeline {
  agent {
    kubernetes {
      yamlFile 'yaml-templates/node-skaffold.yml'
    }
  }
  stages {
    stage('Clone & Install') {
      steps {
        checkout([
          $class: 'GitSCM',
          branches: [
            [name: "*/master"]
          ],
          doGenerateSubmoduleConfigurations: false,
          extensions: [[$class: 'LocalBranch']],, // <-- this double comma
          submoduleCfg: [],
          userRemoteConfigs: [
            [credentialsId: 'redacted',
              url: "redacted"
            ]
          ]
        ])
        sh 'npm ci'
      }
    }
    stage('Build Prod') {
      steps {
        sh 'export APP_ENV=production && npm run build'
      }
    }
  }
}
```

Looks like the double comma was causing the issue, even though it parsed fine to run with (double comma was not needed in fact)

**Comment by nvuillam** [comment]
NglParseError is for... parse errors ^^
When Groovy does not succeed to parse a file, most of issues are not catched
Once the parse errors fixed, the linter can parse the code so provide usual errors :)
